ee/server/service/teams.go

```diff
@@ -404,10 +405,16 @@ func (svc Service) ApplyTeamSpecs(ctx context.Context, specs []*fleet.TeamSpec)
 				if agentOptions == nil {
 					agentOptions = config.AgentOptions
 				}
+				features := spec.Features
```

Comment:
Nit (no need to refactor this now at all, just a style comment for future code, "long" `for` loops like this are hard to follow):
```go
for _, spec := range specs {
  team, err := svc.ds.TeamByName(ctx, spec.Name)
  if err != nil {
    if err := ctxerr.Cause(err); err == sql.ErrNoRows {
      // call new method createTeamFromSpec()
    }
    return err
  }
  // call new method editTeamFromSpec()
}
```

Reply:
since I was there and moving stuff I took the chance to do this, I suspect we'll be adding a ton of functionality here.